Client/mods/deathmatch/logic/CClientGifDisplay.cpp

```diff
@@ -0,0 +1,95 @@
+/*****************************************************************************
+ *
+ *  PROJECT:     Multi Theft Auto
+ *  LICENSE:     See LICENSE in the top level directory
+ *
+ *  Multi Theft Auto is available from https://www.multitheftauto.com/
+ *
+ *****************************************************************************/
+
+#include "StdInc.h"
+#include "CClientGifDisplay.h"
+#include "CClientGif.h"
+
+CClientGifDisplay::CClientGifDisplay(CClientDisplayManager* pDisplayManager, CClientGif* pGif, int ID)
+    : CClientDisplay(pDisplayManager, ID)
+{
+    m_pGif = pGif;
+    m_bVisible = true;
+    m_bIsCleared = false;
+}
+
+void CClientGifDisplay::Render(){
+    if (!m_pGif || m_pGif->IsDestoryed()){
+        return;
+    }
+    if (!m_bVisible){
+        if (!IsCleared()){
+            ClearTexture();
+        }
+        return;
+    }else if (IsCleared()){
+        m_bIsCleared = false;
+    }
+    if (m_pGif->IsPlaying()) {
+        if ((double)GetTickCount64_() >= m_pGif->GetTick() + m_pGif->GetFrameDelay(-1)) {
+            m_pGif->Next();
+            m_pGif->UpdateTick();
+            UpdateTexture();
+        }
+    }
+}
+
+void CClientGifDisplay::UpdateTexture(){
+    if (!m_pGif || m_pGif->IsDestoryed()){
+        return;
+    }
+    CGifItem* pGifItem = m_pGif->GetRenderItem();
+    if (!pGifItem){
+        return;
+    }
+    IDirect3DSurface9* surface = pGifItem->m_pD3DRenderTargetSurface;
+    if (!surface){
+        return;
+    }
+    IDirect3DDevice9* device = pGifItem->m_pDevice;
+    // Lock Surface
+    D3DLOCKED_RECT LockedRect;
+    ZeroMemory(&LockedRect, sizeof(D3DLOCKED_RECT));
+    surface->LockRect(&LockedRect, nullptr, 0);
```

Comment:
You should always check if a texture is actually locked. See PR #2832.

Reply:
fixed this one!